includes/faveo-pricing-functions.php

```diff
@@ -1,400 +1,260 @@
+        $ipList = explode( ',', sanitize_text_field( wp_unslash( $_SERVER['HTTP_X_FORWARDED_FOR'] ) ) );
+        $ip     = trim( $ipList[0] );
+    } elseif ( isset( $_SERVER['REMOTE_ADDR'] ) && ! empty( $_SERVER['REMOTE_ADDR'] ) ) {
+        $ip = sanitize_text_field( wp_unslash( $_SERVER['REMOTE_ADDR'] ) );
+    }
+
+    return $ip;
+}
+
+
+// Get country from IP using ipapi.co (secure HTTPS)
+function fhai_get_country_from_ip($ip) {
+    // Fallback for localhost testing
+    if ($ip === '127.0.0.1' || $ip === '::1') {
+        return 'US';
+    }
+
+    $response = wp_remote_get("https://ipapi.co/{$ip}/json/");
+    if (is_wp_error($response)) {
+        return 'US'; // fallback
+    }
+
+    $data = json_decode(wp_remote_retrieve_body($response), true);
+
+    // ipapi.co returns "country" instead of "countryCode"
+    return !empty($data['country']) ? strtoupper($data['country']) : 'US';
+}
+
+// Currency symbol mapping by code
+function fhai_currency_symbol_by_code($currency_code) {
+    $map = array(
+        'USD' => '$', 'INR' => '₹', 'GBP' => '£', 'EUR' => '€',
+        'CAD' => 'C$', 'AUD' => 'A$', 'NZD' => 'NZ$', 'JPY' => '¥',
+        'CNY' => '¥', 'SGD' => 'S$', 'HKD' => 'HK$', 'KRW' => '₩',
+        'RUB' => '₽', 'BRL' => 'R$', 'ZAR' => 'R', 'MXN' => 'MX$',
+        'CHF' => 'CHF', 'SEK' => 'kr', 'NOK' => 'kr', 'DKK' => 'kr',
+        'PLN' => 'zł', 'CZK' => 'Kč', 'HUF' => 'Ft', 'TRY' => '₺',
+        'SAR' => '﷼', 'AED' => 'د.إ', 'ILS' => '₪', 'THB' => '฿',
+        'IDR' => 'Rp', 'MYR' => 'RM', 'PHP' => '₱', 'VND' => '₫',
+        'NGN' => '₦'
+    );
+    return isset($map[$currency_code]) ? $map[$currency_code] : '$';
+}
+
+// Currency symbol mapping by country
+function fhai_currency_symbol($country_code) {
+    $map = array(
+        'US' => '$', 'IN' => '₹', 'GB' => '£', 'EU' => '€', 'CA' => 'C$',
+        'AU' => 'A$', 'NZ' => 'NZ$', 'JP' => '¥', 'CN' => '¥', 'SG' => 'S$',
+        'HK' => 'HK$', 'KR' => '₩', 'RU' => '₽', 'BR' => 'R$', 'ZA' => 'R',
+        'MX' => 'MX$', 'CH' => 'CHF', 'SE' => 'kr', 'NO' => 'kr', 'DK' => 'kr',
+        'PL' => 'zł', 'CZ' => 'Kč', 'HU' => 'Ft', 'TR' => '₺', 'SA' => '﷼',
+        'AE' => 'د.إ', 'IL' => '₪', 'TH' => '฿', 'ID' => 'Rp', 'MY' => 'RM',
+        'PH' => '₱', 'VN' => '₫', 'NG' => '₦'
+    );
+    return isset($map[$country_code]) ? $map[$country_code] : '$';
+}
+
+// Format Indian number
+function indian_number_format($number) {
+    $number = (string) round($number);
+    $len = strlen($number);
+    if ($len <= 3) return $number;
+    $num = substr($number, -3);
+    $remainder = substr($number, 0, $len - 3);
+    $formatted = '';
+    while (strlen($remainder) > 2) {
+        $formatted = ',' . substr($remainder, -2) . $formatted;
+        $remainder = substr($remainder, 0, -2);
+    }
+    return $remainder . $formatted . ',' . $num;
+}
+
+// Shortcode
 function fhai_calling($atts) {
```

Comment:
This function can be refactored, a lot of reapting code. Hard coded stuff. And more nested and unnecessary loops you use the slower the code gets. We can even create file for these html templates and keep the logic clean